taxi/tests/test_views.py

```diff
@@ -0,0 +1,111 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Driver, Car
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+DRIVER_URL = reverse("taxi:driver-list")
+CAR_URL = reverse("taxi:car-list")
+
+
+class PublicTest(TestCase):
+    def test_login_required_for_manufacturer_list_page(self):
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_for_driver_list_page(self):
+        response = self.client.get(DRIVER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_for_car_list_page(self):
+        response = self.client.get(CAR_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Manufacturer.objects.create(name="BMW", country="Germany")
+        Manufacturer.objects.create(name="Volvo", country="Sweden")
+        cls.car1 = Car.objects.create(
+            model="M4",
+            manufacturer=Manufacturer.objects.get(id=1)
```

Comment:
It's not safe to rely on the id of the objects in the database, especially in tests. The id of an object can change depending on the order of the tests and other factors. Instead of using `id=1`, you can assign the created `Manufacturer` to a variable and use it here.